GPU ERROR issue [possibly CUDA 7.5]

This text is copied out of https://github.com/fireice-uk/xmr-stak/pull/28#pullrequestreview-68650125

The compile still fails, but we are making progress. I googled for a fix and adding `add_definitions(-D_MWAITXINTRIN_H_INCLUDED)` seems to fix the problem.

```
/usr/lib/gcc/x86_64-linux-gnu/5/include/mwaitxintrin.h(36): error: identifier "__builtin_ia32_monitorx" is undefined

/usr/lib/gcc/x86_64-linux-gnu/5/include/mwaitxintrin.h(42): error: identifier "__builtin_ia32_mwaitx" is undefined

2 errors detected in the compilation of "/tmp/tmpxft_00007a1d_00000000-22_cuda_extra.compute_20.cpp1.ii".
CMake Error at xmrstak_cuda_backend_generated_cuda_extra.cu.o.cmake:266 (message):
  Error generating file
  /home/fireice/github/xmr-stak/build/CMakeFiles/xmrstak_cuda_backend.dir/xmrstak/backend/nvidia/nvcc_code/./xmrstak_cuda_backend_generated_cuda_extra.cu.o
```

Can you make sure if that flag needs to be targeted at GCC or not.

-----------------------------------------------------
added by psychocrypt: This error will only occur if the unsupported compiler gcc 5.X is enabled by manually editing of the file `host_config.h`

@fireice-uk Could it be that you changed your CUDA headers? Normally CUDA 7.5 not support gcc 5.X and should crash with the error message

```
include/host_config.h:115:2: error: #error -- unsupported GNU version! gcc versions later than 4.9 are not supported!
 #error -- unsupported GNU version! gcc versions later than 4.9 are not supported!
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -142,7 +142,11 @@ if(CUDA_ENABLE)
             if(CUDA_KEEP_FILES)
                 set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS}" --keep --keep-dir "${PROJECT_BINARY_DIR}")
             endif(CUDA_KEEP_FILES)
-
+        
+            if(CUDA_VERSION VERSION_LESS 8.0)
+                # for CUDA 7.5 fix compile error: https://github.com/fireice-uk/xmr-stak/issues/34
+                set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS}" "-D_MWAITXINTRIN_H_INCLUDED")
+            endif()
         else()
             message(FATAL_ERROR "selected CUDA compiler '${CUDA_COMPILER}' is not supported")
         endif()

